src/Native/System.Native/pal_process.cpp

```diff
@@ -346,3 +356,75 @@ int32_t WTermSig(int32_t status)
 {
     return WTERMSIG(status);
 }
+
+static
+int32_t ConvertPathConfNameToPlatform(PathConfNames& name)
+{
+    switch (name)
```

Comment:
If you decide not to inline (I don't mind either way), it would be prudent not to pass that -1 on to pathconf in the error case. I'd put a check and error out as we do for the memory map options for example.